kuma/javascript/src/header/login.jsx

```diff
@@ -37,6 +37,7 @@ export default function Login(): React.Node {
             </>
         );
         let viewProfileLink = `/${locale}/profiles/${userData.username}`;
+        let contributionsLink = `/${locale}/dashboards/revisions?user=${userData.username}`;
```

Comment:
Does it work if your username has a space in it? (...if that's even possible to have!)
Does it work if your username has a non-ascii or a control character in it?

Reply:
I know there's a LOT of users with usernames that `[^a-z0-9_-]` in them. E.g. `Peter Wennblom` and `madeup@gmail.com`.
I think the safest thing is to use something like 
```suggestion
        let contributionsLink = `/${locale}/dashboards/revisions?user=${encodeURIComponent(userData.username)}`;
```
...but I haven't tested it.

Reply:
This is how we currently do it on the Jinja side:

```
<a href="{{ url('dashboards.revisions') }}?user={{ detail_user.username }}" class="button">{{ _('View all activity') }}</a>
```

Reply:
Shall I go ahead and encode the username here @peterbe?

Reply:
Yes please. You never know what might go wrong if you don't encode it. I think it's safest to do so. For example, what if you come up with a crafty username that causes a URL that does unintentional things we couldn't have foreseen. Security or just breakage, either way it's easy to prevent. 

Had you done this:
```javascript
let sp = new URLSearchParams();
sp.set('username', userData.username);
let contributionsLink = `/${locale}/dashboards/revisions?${sp.toString()}`;
```
that would have produced a string of `/en-US/dashboards/revisions?user=p%20%C3%A9ter%40gmail.com`. And you'd get exactly the same result with:

```javascript
let contributionsLink = `/${locale}/dashboards/revisions?user=${encodeURIComponent(
    userData.username
)}`;
```
I checked and confirmed that. 

For the record, I tested it all by changing my username on the Django shell to `u.username = 'p éter@gmail.com'` and things still worked without encoding:
<img width="1480" alt="Screen Shot 2020-06-03 at 4 25 06 PM" src="https://user-images.githubusercontent.com/26739/83685985-7b08e480-a5b7-11ea-9781-8f0e8472a6df.png">